sdks/java/core/src/main/java/org/apache/beam/sdk/io/UnboundedSource.java

```diff
@@ -203,6 +224,11 @@ public byte[] getCurrentRecordId() throws NoSuchElementException {
       return EMPTY;
     }
 
+    /* Returns the offset for the current record of this unbounded reader. */
+    public byte[] getCurrentRecordOffset() {
+      return EMPTY;
```

Comment:
see other comment, seems we could throw like the default checkpoint method if we guard at caller

Reply:
Done.